mlflow/utils/search_utils.py

```diff
@@ -607,16 +655,47 @@ def _parse_filter_for_model_registry(cls, filter_string, valid_search_keys):
                 error_code=INVALID_PARAMETER_VALUE,
             )
         statement = parsed[0]
-        invalids = list(filter(cls._invalid_statement_token, statement.tokens))
+        invalids = list(
+            filter(cls._invalid_statement_token_search_model_registry, statement.tokens)
+        )
         if len(invalids) > 0:
             invalid_clauses = ", ".join("'%s'" % token for token in invalids)
             raise MlflowException(
                 "Invalid clause(s) in filter string: %s. " "%s" % (invalid_clauses, expected),
                 error_code=INVALID_PARAMETER_VALUE,
             )
+        # process non-Comparison tokens
+        if len(statement.tokens) == 1 and isinstance(statement.tokens[0], Comparison):
+            token_list = statement.tokens
+        elif len(statement.tokens) > 1:
```

Comment:
what happens if `len(statement.tokens) == 1 and not instance(statement.tokens[0], Comparison)`?

Reply:
also is it possible for `len(statement.tokens) == 0` to occur?

Reply:
@the first comment - great point - I missed that, and I think that may have been an existing bug too, so will add a case for it (though tbh, I'm not sure if it's very likely that kind of query can be constructed.

@the second comment - I don't think it's possible but I've added an error message to signal that